src/components/translatable/TranslatableTSV.js

```diff
@@ -101,11 +103,45 @@ export default function TranslatableTSV({
     return rowId;
   }, [columnNames, delimiters]);
 
-  const options = {
+  const [options, setOptions] = useState({
     page: 0,
     rowsPerPage: 25,
     rowsPerPageOptions: [10, 25, 50, 100],
-  };
+  });
+
+  const { query } = usePermalinks({});
+
+  useEffect(() => {
```

Comment:
This file is already pretty large and we're trying to keep it from growing. Can we abstract the logic to another file such as a helpers.js file to house the logic and call them in the useDeepCompareEffects?